packages/superset-ui-chart/src/components/SuperChart.tsx

```diff
@@ -0,0 +1,222 @@
+import * as React from 'react';
+import { createSelector } from 'reselect';
+import getChartComponentRegistry from '../registries/ChartComponentRegistrySingleton';
+import getChartTransformPropsRegistry from '../registries/ChartTransformPropsRegistrySingleton';
+import ChartProps from '../models/ChartProps';
+import createLoadableRenderer, { LoadableRenderer } from './createLoadableRenderer';
+
+const IDENTITY = (x: any) => x;
+
+const EMPTY = () => null;
+
+/* eslint-disable sort-keys */
+const defaultProps = {
+  id: '',
+  className: '',
+  preTransformProps: IDENTITY,
+  overrideTransformProps: undefined,
+  postTransformProps: IDENTITY,
+  onRenderSuccess() {},
+  onRenderFailure() {},
+};
+/* eslint-enable sort-keys */
+
+type TransformFunction<Input = PlainProps, Output = PlainProps> = (x: Input) => Output;
+type HandlerFunction = (...args: any[]) => void;
+
+interface LoadingProps {
+  error: any;
+}
+
+interface PlainProps {
+  [key: string]: any;
+}
+
+interface LoadedModules {
+  Chart: React.Component | { default: React.Component };
+  transformProps: TransformFunction | { default: TransformFunction };
+}
+
+interface RenderProps {
+  chartProps: ChartProps;
+  preTransformProps?: TransformFunction<ChartProps>;
+  postTransformProps?: TransformFunction;
+}
+
+const BLANK_CHART_PROPS = new ChartProps();
+
+export interface SuperChartProps {
+  id?: string;
+  className?: string;
+  chartProps?: ChartProps | null;
+  chartType: string;
+  preTransformProps?: TransformFunction<ChartProps>;
+  overrideTransformProps?: TransformFunction;
```

Comment:
Should both `overrideTransformProps` and `postTransformProps ` take `ChartProps` as type arguments?

Reply:
If `preTransformProps` transforms `ChartProps` into something else, that will be input to `overrideTransformProps` and/or `postTransformProps`

Reply:
That's true. I suppose we could create a generic over the output props and use that as the input for `overrideTransformProps` and `postTransformProps`.

I'm cool leaving this as-is for now (perhaps adding a TODO to revisit?). Think we could potentially improve the types here once we have more chart plugins added.